src/apis/stockx/asks/index.js

```diff
@@ -0,0 +1,140 @@
+import moment from 'moment';
+
+import Api from '../../base';
+import { randomInclusive } from '../../../utils';
+
+export default class AsksApi extends Api {
+  // TODO!
+  async list(options = {}) {}
+
+  async update(ask = {}, options = {}) {
+    const { amount } = options;
+    const { headers, bearer, jar, proxy, currency } = this.data;
+    const expiresAt = moment().add(30, 'days').utc().format();
+
+    let chainId;
+    let skuUuid;
+    try {
+      ({ chainId, skuUuid } = ask);
+
+      if (!amount || !skuUuid || !chainId) {
+        const error = new Error('Invalid amount product id, and/or ask id!');
+        error.status = 404;
+        throw error;
+      }
+
+      if (!bearer) {
+        const error = new Error('Please login first!');
+        error.status = 401;
+        throw error;
+      }
+
+      const res = await this._request('https://stockx.com/api/portfolio?a=ask', {
+        headers: {
+          ...headers,
+          authorization: `Bearer ${bearer}`,
+          'content-type': 'application/json',
+        },
+        jar,
```

Comment:
Given this is duplicated for each request, would be good to abstract it out to a common call, abstracting the request might be a cleaner approach which just takes optional arguments to suppliment the headers, body, etc.

Reply:
Good thinking. I think for this PR this is fine, we can do a follow up PR to cleanup and condense.